plugin/pkg/admission/initialresources/admission_test.go

```diff
@@ -112,15 +113,40 @@ func expectNoAnnotation(t *testing.T, pod *api.Pod) {
 func admit(t *testing.T, ir admission.Interface, pods []*api.Pod) {
 	for i := range pods {
 		p := pods[i]
-		if err := ir.Admit(admission.NewAttributesRecord(p, nil, api.Kind("Pod").WithVersion("version"), "test", p.ObjectMeta.Name, api.Resource("pods").WithVersion("version"), "", admission.Create, nil)); err != nil {
+
+		podKind := api.Kind("Pod").WithVersion("version")
+		podRes := api.Resource("pods").WithVersion("version")
+		attrs := admission.NewAttributesRecord(p, nil, podKind, "test", p.ObjectMeta.Name, podRes, "", admission.Create, nil)
+		if err := ir.Admit(attrs); err != nil {
 			t.Error(err)
 		}
 	}
 }
 
+func testAdminReturnsWhenSubresourceIsSet(t *testing.T, ir admission.Interface, p *api.Pod) {
+	podKind := api.Kind("Pod").WithVersion("version")
+	podRes := api.Resource("pods").WithVersion("version")
+	attrs := admission.NewAttributesRecord(p, nil, podKind, "test", p.ObjectMeta.Name, podRes, "foo", admission.Create, nil)
+	if err := ir.Admit(attrs); err != nil {
+		t.Error(err)
+	}
+}
+
+func testAdminFailsWhenTypeNotPod(t *testing.T, ir admission.Interface) {
```

Comment:
Can we combine those test funcs into one, but with different test cases? for example:

```
test:=[]struct{
  input
  expected
}{
}

err := ir.Admin(input); 
...
```